libraries/app/database_api.cpp

```diff
@@ -1087,6 +1088,32 @@ vector<call_order_object> database_api_impl::get_margin_positions( const account
    } FC_CAPTURE_AND_RETHROW( (id) )
 }
 
+vector<collateral_bid_object> database_api::get_collateral_bids(const asset_id_type asset, uint32_t limit, uint32_t start)const
+{
+   return my->get_collateral_bids( asset, limit, start );
+}
+
+vector<collateral_bid_object> database_api_impl::get_collateral_bids(const asset_id_type asset_id, uint32_t limit, uint32_t skip)const
+{ try {
+   FC_ASSERT( limit <= 100 );
+   const asset_object& swan = asset_id(_db);
+   FC_ASSERT( swan.is_market_issued() );
```

Comment:
This is just an accessor method. I think it will simplify things if we're lenient here. The call will return an empty list if the requested asset is not in global settlement. (This is technically correct.)